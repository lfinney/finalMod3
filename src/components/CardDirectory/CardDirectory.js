import React from 'react';
import Card from '../Card/Card';
import PropTypes from 'prop-types';

const CardDirectory = ({ housesInTheGame }) => {
  const houseInfo= housesInTheGame.map( (house, index) => <Card house={house} key ={`house - ${index}`}/>);

  return (
    <div>{houseInfo}</div>
  );
};

CardDirectory.propTypes = {
  housesInTheGame: PropTypes.arrayOf(PropTypes.object)
};

export default CardDirectory;
