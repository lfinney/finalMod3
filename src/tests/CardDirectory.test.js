import React from 'react';
import CardDirectory from '../components/CardDirectory/CardDirectory';
import { shallow } from 'enzyme';
import housesMock from './mockData/housesMock';

describe('CardDirectory', () => {

  it('should render component', () => {
    const wrapper = shallow(
      <CardDirectory housesInTheGame={housesMock} />);

    expect(wrapper).toMatchSnapshot();
  });
});
