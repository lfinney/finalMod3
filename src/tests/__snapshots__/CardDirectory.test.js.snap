// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardDirectory should render component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardDirectory
    housesInTheGame={
        Array [
            Object {
              "ancestralWeapons": Array [
                "Lady Forlorn",
              ],
              "coatOfArms": "Three black ravens in flight...",
              "diedOut": "",
              "founded": "",
              "name": "House Corbray of Heart's Home",
              "seats": Array [
                "Heart's Home",
              ],
              "swornMembers": Array [
                "Jon Snow",
              ],
              "titles": Array [
                "Lord of Heart's Home",
                "King of the Fingers (historical)",
              ],
            },
            Object {
              "ancestralWeapons": Array [
                "Dawn",
              ],
              "coatOfArms": "A white sword falling...",
              "diedOut": "",
              "founded": "",
              "name": "House Dayne of Starfall",
              "seats": Array [
                "Starfall",
              ],
              "swornMembers": Array [
                "Tyrion Lannister",
              ],
              "titles": Array [
                "Lord of Starfall",
                "King of the Torrentine (historical)",
              ],
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          house={
                    Object {
                              "ancestralWeapons": Array [
                                "Lady Forlorn",
                              ],
                              "coatOfArms": "Three black ravens in flight...",
                              "diedOut": "",
                              "founded": "",
                              "name": "House Corbray of Heart's Home",
                              "seats": Array [
                                "Heart's Home",
                              ],
                              "swornMembers": Array [
                                "Jon Snow",
                              ],
                              "titles": Array [
                                "Lord of Heart's Home",
                                "King of the Fingers (historical)",
                              ],
                            }
          }
/>,
        <Card
          house={
                    Object {
                              "ancestralWeapons": Array [
                                "Dawn",
                              ],
                              "coatOfArms": "A white sword falling...",
                              "diedOut": "",
                              "founded": "",
                              "name": "House Dayne of Starfall",
                              "seats": Array [
                                "Starfall",
                              ],
                              "swornMembers": Array [
                                "Tyrion Lannister",
                              ],
                              "titles": Array [
                                "Lord of Starfall",
                                "King of the Torrentine (historical)",
                              ],
                            }
          }
/>,
      ],
      "className": "card-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "house - 0",
        "nodeType": "class",
        "props": Object {
          "house": Object {
            "ancestralWeapons": Array [
              "Lady Forlorn",
            ],
            "coatOfArms": "Three black ravens in flight...",
            "diedOut": "",
            "founded": "",
            "name": "House Corbray of Heart's Home",
            "seats": Array [
              "Heart's Home",
            ],
            "swornMembers": Array [
              "Jon Snow",
            ],
            "titles": Array [
              "Lord of Heart's Home",
              "King of the Fingers (historical)",
            ],
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "house - 1",
        "nodeType": "class",
        "props": Object {
          "house": Object {
            "ancestralWeapons": Array [
              "Dawn",
            ],
            "coatOfArms": "A white sword falling...",
            "diedOut": "",
            "founded": "",
            "name": "House Dayne of Starfall",
            "seats": Array [
              "Starfall",
            ],
            "swornMembers": Array [
              "Tyrion Lannister",
            ],
            "titles": Array [
              "Lord of Starfall",
              "King of the Torrentine (historical)",
            ],
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            house={
                        Object {
                                    "ancestralWeapons": Array [
                                      "Lady Forlorn",
                                    ],
                                    "coatOfArms": "Three black ravens in flight...",
                                    "diedOut": "",
                                    "founded": "",
                                    "name": "House Corbray of Heart's Home",
                                    "seats": Array [
                                      "Heart's Home",
                                    ],
                                    "swornMembers": Array [
                                      "Jon Snow",
                                    ],
                                    "titles": Array [
                                      "Lord of Heart's Home",
                                      "King of the Fingers (historical)",
                                    ],
                                  }
            }
/>,
          <Card
            house={
                        Object {
                                    "ancestralWeapons": Array [
                                      "Dawn",
                                    ],
                                    "coatOfArms": "A white sword falling...",
                                    "diedOut": "",
                                    "founded": "",
                                    "name": "House Dayne of Starfall",
                                    "seats": Array [
                                      "Starfall",
                                    ],
                                    "swornMembers": Array [
                                      "Tyrion Lannister",
                                    ],
                                    "titles": Array [
                                      "Lord of Starfall",
                                      "King of the Torrentine (historical)",
                                    ],
                                  }
            }
/>,
        ],
        "className": "card-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "house - 0",
          "nodeType": "class",
          "props": Object {
            "house": Object {
              "ancestralWeapons": Array [
                "Lady Forlorn",
              ],
              "coatOfArms": "Three black ravens in flight...",
              "diedOut": "",
              "founded": "",
              "name": "House Corbray of Heart's Home",
              "seats": Array [
                "Heart's Home",
              ],
              "swornMembers": Array [
                "Jon Snow",
              ],
              "titles": Array [
                "Lord of Heart's Home",
                "King of the Fingers (historical)",
              ],
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "house - 1",
          "nodeType": "class",
          "props": Object {
            "house": Object {
              "ancestralWeapons": Array [
                "Dawn",
              ],
              "coatOfArms": "A white sword falling...",
              "diedOut": "",
              "founded": "",
              "name": "House Dayne of Starfall",
              "seats": Array [
                "Starfall",
              ],
              "swornMembers": Array [
                "Tyrion Lannister",
              ],
              "titles": Array [
                "Lord of Starfall",
                "King of the Torrentine (historical)",
              ],
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
